core/config/engine.h

```diff
@@ -36,6 +36,11 @@
 #include "core/templates/list.h"
 #include "core/templates/vector.h"
 
+#if defined(VULKAN_ENABLED) && (defined(DEBUG_ENABLED) || defined(DEV_ENABLED))
```

Comment:
Should be moved to `drivers/vulkan`.

Reply:
I'm looking into this PR.

It seems like it can't be moved there.

It appears that the `VK_TRACK_DEVICE_MEMORY` is a debug-only macro (it's only defined in debug & dev builds).

When that macro is defined, objects from core inform the Vulkan subsystem what subsystem they belong to.

e.g. in core/core_bind.cpp we have things like:

```cpp
#if defined(VK_TRACK_DRIVER_MEMORY) || defined(VK_TRACK_DEVICE_MEMORY)
String Engine::get_tracked_object_name(uint32_t typeIndex) const {
	return RenderingContextDriverVulkan::get_tracked_object_name(typeIndex);
}
uint64_t Engine::get_tracked_object_type_count() const {
	return RenderingContextDriverVulkan::get_tracked_object_type_count();
}
#endif
```

Where get_tracked_object_name returns a string like "INSTANCE", "PHYSICAL_DEVICE", "DESCRIPTOR_SET_LAYOUT".

I don't know if there is a better way yet, but it's not a simple thing of moving that line to a different header.